server.go

```diff
@@ -495,8 +507,10 @@ func (s *server) inboundPeerConnected(conn net.Conn) {
 	// cancel the ongoing connection attempts to ensure that we don't end
 	// up with a duplicate connecting to the same peer.
 	s.pendingConnMtx.RLock()
-	if connReq, ok := s.persistentConnReqs[pubStr]; ok {
-		s.connMgr.Remove(connReq.ID())
+	if connReqs, ok := s.persistentConnReqs[pubStr]; ok {
+		for _, connReq := range connReqs {
```

Comment:
A similar code block (removing all the connection requests for the particular node pubkey) needs to be inserted below within the `outboundPeerConnected` method. We'll need to similarly cancel all the pending outbound requests (launched within `newServer`) as we've successfully established a connection.